devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1037,6 +1201,108 @@ void HumanStateProvider::run()
     //                                          pImpl->linkErrorAngularVelocities);
 }
 
+bool HumanStateProvider::impl::applyRpcCommand()
+{
+    // check is the choosen links are valid
+    std::string linkName = commandPro->parentLinkName;
+    std::string childLinkName = commandPro->childLinkName;
+    if (!(linkName == "") && (wearableStorage.modelToWearable_LinkName.find(linkName) == wearableStorage.modelToWearable_LinkName.end())) {
+        yWarning() << LogPrefix << "link " << linkName << " choosen for secondaty calibration is not valid";
+        return false;
+    }
+    if (!(childLinkName == "") && (wearableStorage.modelToWearable_LinkName.find(childLinkName) == wearableStorage.modelToWearable_LinkName.end())) {
+        yWarning() << LogPrefix << "link " << childLinkName << " choosen for secondaty calibration is not valid";
+        return false;
+    }
+
+    if (commandPro->cmdErease) {
+        if (linkName != "") {
+            secondaryCalibrationRotations.erase(linkName);
+            yInfo() << LogPrefix << "Discarding the secondary calibration matrix for link " << linkName;
+        }
+        else {
+            secondaryCalibrationRotations.clear();
+            yInfo() << LogPrefix << "Discarding all the secondary calibration matrices";
+        }
+    }
+    else {
+        iDynTree::VectorDynSize jointPos;
```

Comment:
how would you suggest to define/allocate it?

Reply:
It depends on whether this function has to be fast or not. If not, it is ok as it is. Otherwise, you can make all the vectors declared here as class members.

Reply:
in this case it is not relevant beeing fast so should be ok, anyway I have improved slightly the code avoiding passing trough `kinDynComputations` (https://github.com/robotology/human-dynamics-estimation/pull/164/commits/766f6b3f9cd7079bbda2814f549685c4f9868175)